### PR TITLE
chore: add rewards coordinator deployment script

### PR DESCRIPTION
**Pull request by shrimalmadhur**
- Add devnet deployment scripts for Rewards Coordinator

### CODE REVIEW

File: script/deploy/devnet/M2_Deploy_From_Scratch.s.sol [hunk 1]
```diff
@@ -88,6 +91,17 @@ contract Deployer_M2 is Script, Test {
     uint256 DELEGATION_WITHDRAWAL_DELAY_BLOCKS;
     uint256 EIGENPOD_MANAGER_INIT_PAUSED_STATUS;
     uint256 DELAYED_WITHDRAWAL_ROUTER_INIT_PAUSED_STATUS;
+    uint256 REWARDS_COORDINATOR_INIT_PAUSED_STATUS;
```
Comment (samlaf) [inline script/deploy/devnet/M2_Deploy_From_Scratch.s.sol:94]: move it down with the other RewardsCoordinator vars?

File: script/deploy/devnet/M2_Deploy_From_Scratch.s.sol [hunk 2]
```diff
@@ -112,6 +126,25 @@ contract Deployer_M2 is Script, Test {
...
+        REWARDS_COORDINATOR_MAX_FUTURE_LENGTH = uint32(stdJson.readUint(config_data, ".rewardsCoordinator.MAX_FUTURE_LENGTH"));
+        REWARDS_COORDINATOR_GENESIS_REWARDS_TIMESTAMP = uint32(stdJson.readUint(config_data, ".rewardsCoordinator.GENESIS_REWARDS_TIMESTAMP"));
+        REWARDS_COORDINATOR_UPDATER = stdJson.readAddress(config_data, ".rewardsCoordinator.rewards_updater_address");
+        REWARDS_COORDINATOR_ACTIVATION_DELAY = uint32(stdJson.readUint(config_data, ".rewardsCoordinator.activation_delay"));
+        REWARDS_COORDINATOR_CALCULATION_INTERVAL_SECONDS = uint32(
+            stdJson.readUint(config_data, ".rewardsCoordinator.calculation_interval_seconds")
+        );
+        REWARDS_COORDINATOR_GLOBAL_OPERATOR_COMMISSION_BIPS = uint32(
+            stdJson.readUint(config_data, ".rewardsCoordinator.global_operator_commission_bips")
+        );
```
Comment (samlaf) [inline script/deploy/devnet/M2_Deploy_From_Scratch.s.sol:147]: make all vars (also in the json) same cased. all caps probably?

**Review by samlaf — APPROVED** [review]
LGTM

File: script/deploy/devnet/M2_Deploy_From_Scratch.s.sol (hunk 1 above)
Comment (shrimalmadhur) [reply script/deploy/devnet/M2_Deploy_From_Scratch.s.sol:93]: moved the delayed box below

File: script/deploy/devnet/M2_Deploy_From_Scratch.s.sol (hunk 2 above)
Comment (shrimalmadhur) [reply script/deploy/devnet/M2_Deploy_From_Scratch.s.sol:147]: skipping it for now

**Pull request merged**